### DISCUSSION TITLE
ExecutorPoolCaptureOom maximumPoolSize 999?

### DISCUSSION THREAD

**Discussion by hxl045** [apache/kyuubi#1585] [Q&A]
I try to submit 100 concurrent beeline xxx:10009 tasks,then..just 40 can be executed,yarn mem&&cpu resources are sufficient。
![1639726834](https://user-images.githubusercontent.com/50735118/146507749-d23b627d-dfb5-493c-af09-670a6fa0e1fe.jpg)

**Comment by pan3793** [comment]
Try enlarge `kyuubi.frontend.thrift.max.worker.threads`, which default value is 999.
BTW, please paste log text instead of picture.

**Reply by hxl045** [reply]
Got it,tks!!!

**Comment by yaooqinn** [comment]
thanks for trying，how many concurrent users and applications do you have? we can increase the thead pool size for both FE and BE of single server instance, and also use more server instances for load balancing to increase the total client concurrency.

BR,

Kent

**Reply by hxl045** [reply]
Thanks for your advice！
Just test users~
